tests/test_workflow_opencourseware.py

```diff
@@ -0,0 +1,240 @@
+# ruff: noqa: B015, SLF001
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+INSTRUCTORS = [
+    {
+        "first_name": "Kerry",
+        "last_name": "Oki",
+        "middle_initial": "",
+        "salutation": "Prof.",
+        "title": "Prof. Kerry Oki",
+    },
+    {
+        "first_name": "Earl",
+        "last_name": "Bird",
+        "middle_initial": "E.",
+        "salutation": "Prof.",
+        "title": "Prof. Earl E. Bird",
+    },
+]
+
+
+@patch("dsc.workflows.opencourseware.OpenCourseWare._extract_metadata_from_zip_file")
+@patch("dsc.utilities.aws.s3.S3Client.files_iter")
+def test_workflow_ocw_reconcile_bitstreams_and_metadata_success(
+    mock_s3_client_files_iter,
+    mock_opencourseware_extract_metadata_from_zip_file,
+    opencourseware_source_metadata,
+    opencourseware_workflow_instance,
+):
+    mock_s3_client_files_iter.return_value = [
+        "s3://dsc/opencourseware/batch-aaa/123.zip",
+        "s3://dsc/opencourseware/batch-aaa/124.zip",
+    ]
+    opencourseware_source_metadata["instructors"] = "Edelman, Alan|Johnson, Steven G."
+    mock_opencourseware_extract_metadata_from_zip_file.side_effect = [
+        opencourseware_source_metadata,
+        FileNotFoundError,
+    ]
+    assert opencourseware_workflow_instance.reconcile_bitstreams_and_metadata() == (
+        set(),
+        {"124"},
+    )
+
+
+def test_workflow_ocw_build_bitstream_dict_success(
+    mocked_s3, opencourseware_workflow_instance, s3_client
+):
+    s3_client.put_file(
+        file_content="",
+        bucket="dsc",
+        key="opencourseware/batch-aaa/123.zip",
+    )
+    s3_client.put_file(
+        file_content="",
+        bucket="dsc",
+        key="opencourseware/batch-aaa/124.zip",
+    )
+    s3_client.put_file(
+        file_content="",
+        bucket="dsc",
+        key="opencourseware/batch-aaa/ignore_me.txt",
+    )
+    assert opencourseware_workflow_instance._build_bitstream_dict() == {
+        "123": ["opencourseware/batch-aaa/123.zip"],
+        "124": ["opencourseware/batch-aaa/124.zip"],
+    }
+
+
+@patch("dsc.workflows.opencourseware.OpenCourseWare._extract_metadata_from_zip_file")
+def test_workflow_ocw_identify_bitstreams_with_metadata_success(
+    mock_opencourseware_extract_metadata_from_zip_file,
+    opencourseware_source_metadata,
+    opencourseware_workflow_instance,
+):
+    opencourseware_source_metadata["instructors"] = "Edelman, Alan|Johnson, Steven G."
+    mock_opencourseware_extract_metadata_from_zip_file.side_effect = [
+        opencourseware_source_metadata,
+        FileNotFoundError,
+    ]
+
+    assert opencourseware_workflow_instance._identify_bitstreams_with_metadata(
+        item_identifiers=["123", "124"]
+    ) == ["123"]
+
+
+@patch("dsc.workflows.opencourseware.OpenCourseWare._read_metadata_json_file")
```

Comment:
Both of these tests could be a lot simpler with 2 zip file fixtures (one with metadata, one without) that you load into `mocked_s3` with the `s3_client` fixture as you did in the `test_workflow_ocw_build_bitstream_dict_success`

Reply:
Changes made. See [updates](https://github.com/MITLibraries/dspace-submission-composer/blob/IN-1096-ocw-workflow/tests/test_workflow_opencourseware.py#L77-L106).